### PR TITLE
feat: reusable build test release

### PR DESCRIPTION
**Pull request by uoboda-splunk**
test runs:
https://github.com/splunk/splunk-add-on-for-jira-cloud/runs/4447430176?check_suite_focus=true
https://github.com/splunk/splunk-add-on-for-servicenow/runs/4453863245?check_suite_focus=true

### CODE REVIEW

File: .github/workflows/reusable-build-test-release.yml [hunk 1]
```diff
@@ -0,0 +1,1328 @@
...
+          # shellcheck disable=SC2157
+          if [ -z "${{ steps.retry-wf.outputs.workflow-name }}" ]; then
+            WORKFLOW_NAME=${{ steps.run-tests.outputs.workflow-name }}
+          else
+            WORKFLOW_NAME="${{ steps.retry-wf.outputs.workflow-name }}"
+          fi
+          echo "pulling logs"
+          mkdir -p ${{ needs.setup.outputs.directory-path }}/argo-logs
+          aws s3 cp s3://${{ needs.setup.outputs.s3-bucket }}/${WORKFLOW_NAME}/ ${{ needs.setup.outputs.directory-path }}/argo-logs/ --recursive
+      - uses: actions/upload-artifact@v2.2.4
```
Comment (arys-splunk) [inline .github/workflows/reusable-build-test-release.yml:712]: Can we use just "v2" here? And probably in other places as well?

Comment (uoboda-splunk) [reply .github/workflows/reusable-build-test-release.yml:712]: I think we can, but I would prefer not to do it right away. Let's actually use this workflow (exactly the same as we use in repos now) and then if we do not observe any problems we can bump up versions

Comment (arys-splunk) [reply .github/workflows/reusable-build-test-release.yml:712]: okay, makes sense

**Review by arys-splunk — APPROVED** [review]

**Pull request merged**